lib/widgets/action_sheet.dart

```diff
@@ -36,20 +36,46 @@ void showMessageActionSheet({required BuildContext context, required Message mes
       && reactionWithVotes.userIds.contains(store.selfUserId))
     ?? false;
 
-  showDraggableScrollableModalBottomSheet<void>(
+  final designVariables = DesignVariables.of(context);
+  showModalBottomSheet<void>(
```

Comment:
For the current behavior of the action sheet (mentioned in [#90 comment](https://github.com/zulip/zulip-flutter/issues/90#issuecomment-2016666599)), there is no need to use `showDraggableScrollableModalBottomSheet`.

As a side note, `_DragHandleLayer` is no longer necessary in `showDraggableScrollableModalBottomSheet` as there is `showDragHandle` property available in `showModalBottomSheet`. Also, `showDraggableScrollableModalBottomSheet` is no longer used in the codebase, so I am not sure if it is meant to stay there for future usage.

Reply:
Yeah, a follow-up commit in this PR can remove that function — that'd be a nice cleanup, thanks. I expect we'll want all our bottom sheets to work the same way as this one.

And if we do want to go back and use that implementation for something, it's always there in the Git history.


> As a side note, `_DragHandleLayer` is no longer necessary in `showDraggableScrollableModalBottomSheet` as there is `showDragHandle` property available in `showModalBottomSheet`.

True. It looks like that was added in https://github.com/flutter/flutter/pull/122445, back in 2023-03… and both @chrisbobbe and I actually reviewed it (mainly on the predecessor thread https://github.com/flutter/flutter/pull/120855). That landed after we merged Chris's implementation here, in #12.

I don't recall if that upstream implementation had behavior differences that would have caused us to stick with ours. I suspect we just didn't bother to switch to it, having already moved on to other tasks by the time it landed upstream.

Reply:
> I don't recall if that upstream implementation had behavior differences that would have caused us to stick with ours. I suspect we just didn't bother to switch to it, having already moved on to other tasks by the time it landed upstream.

Yes, I investigated some months ago and there was a behavior difference that caused us to stick with ours. Here's the context around where we use `_DragHandleLayer`:

```dart
    builder: (BuildContext context) {
      // Make the content start below the drag handle in the y-direction, but
      // when the content is scrollable, let it scroll under the drag handle in
      // the z-direction.
      return Stack(
        children: [
          _DraggableScrollableLayer(builder: builder),
          _DragHandleLayer(),
        ]);
    });
```

If we migrated to `showDragHandle: true`, then we'd have to give up the part after "but" in that comment. You can simulate this by making there be more than a screenful of options in the sheet:

| Current | Migrated to `showDragHandle: true` |
| --- | --- |
| <img width="559" alt="image" src="https://github.com/user-attachments/assets/3d01262a-f151-458a-a7c4-bdc957824434"> | <img width="559" alt="image" src="https://github.com/user-attachments/assets/9655781c-1c7e-41d1-a186-c849e2f2bd75"> |

The culprit is a `Padding` in upstream that always forces the caller's content to begin below the drag-handle area in the y-direction:

<details>
<summary>code</summary>

https://github.com/flutter/flutter/blob/ba31bf8c0/packages/flutter/lib/src/material/bottom_sheet.dart#L402
```dart
    Widget bottomSheet = Material(
      key: _childKey,
      color: color,
      elevation: elevation,
      surfaceTintColor: surfaceTintColor,
      shadowColor: shadowColor,
      shape: shape,
      clipBehavior: clipBehavior,
      child: NotificationListener<DraggableScrollableNotification>(
        onNotification: extentChanged,
        child: !showDragHandle
          ? widget.builder(context)
          : Stack(
              alignment: Alignment.topCenter,
              children: <Widget>[
                dragHandle!,
                Padding(
                  padding: const EdgeInsets.only(top: kMinInteractiveDimension),
                  child: widget.builder(context),
                ),
              ],
            ),
      ),
    );
```
</details>

I did [mention](https://github.com/flutter/flutter/pull/120855#discussion_r1117789100) our desired behavior in my review of that upstream work that Greg mentioned, but didn't follow up when it turned out not to be supported.

Reply:
(This does turn out to be moot for us, though, since the redesigned sheet doesn't have a drag handle.)